src/kernels/jupyter/launcher/liveshare/hostJupyterServer.ts

```diff
@@ -150,6 +151,22 @@ export class HostJupyterServer implements INotebookServer {
         if (!this.sessionManager || this.isDisposed) {
             throw new SessionDisposedError();
         }
+        if (
+            this.sessionManager &&
+            !this.isDisposed &&
+            (kernelConnection.kind === 'connectToLiveRemoteKernel' ||
+                kernelConnection.kind === 'startUsingRemoteKernelSpec')
+        ) {
+            try {
+                await Promise.all([this.sessionManager.getRunningKernels(), this.sessionManager.getKernelSpecs()]);
```

Comment:
Ensure we try to fetch the kernels (validate the remote jupyter server connection) before creating a session.
This should also fix https://github.com/microsoft/vscode-jupyter/issues/8043

Reply:
As well as the current issue #9167